src/__swaps__/screens/Swap/resources/search/searchV2.ts

```diff
@@ -74,22 +95,129 @@ export const useUnverifiedTokenSearchStore = createQueryStore<SearchAsset[], Tok
     fetcher: (params, abortController) => ((params.query?.length ?? 0) > 2 ? searchUnverifiedTokens(params, abortController) : NO_RESULTS),
     transform: (data, { query }) =>
       query && isAddress(query) ? getExactMatches(data, query, true, MAX_UNVERIFIED_RESULTS) : data.slice(0, MAX_UNVERIFIED_RESULTS),
-
     disableAutoRefetching: true,
     keepPreviousData: true,
     params: {
       list: TokenLists.HighLiquidity,
       chainId: $ => $(useSwapsStore).selectedOutputChainId,
-      keys: $ => $(useSwapsSearchStore, state => getSearchKeys(state.searchQuery.trim())),
       query: $ => $(useSwapsSearchStore, state => state.searchQuery.trim()),
-      threshold: $ => $(useSwapsSearchStore, state => getSearchThreshold(state.searchQuery.trim())),
     },
     staleTime: time.minutes(2),
   },
-
   { persistThrottleMs: time.seconds(12), storageKey: 'unverifiedTokenSearch' }
 );
 
+export const useDiscoverSearchStore = createQueryStore<DiscoverSearchResults, DiscoverSearchParams>(
+  {
+    fetcher: (params, abortController) => discoverSearchQueryFunction(params, abortController),
+    cacheTime: params => (params.query?.length ? time.seconds(15) : time.hours(1)),
+    disableAutoRefetching: true,
+    keepPreviousData: true,
+    params: {
+      query: $ => $(useDiscoverSearchQueryStore, state => (state.searchQuery.trim().length ? state.searchQuery.trim() : '')),
+    },
+    staleTime: time.minutes(2),
+  },
+  { persistThrottleMs: time.seconds(8), storageKey: 'discoverTokenSearch' }
+);
+
+const sortForDefaultList = (tokens: SearchAsset[]) => {
+  const curated = tokens.filter(asset => asset.highLiquidity && asset.isRainbowCurated && asset.icon_url);
+  return curated.sort((a, b) => (b.market?.market_cap?.value || 0) - (a.market?.market_cap?.value || 0));
+};
+
+const sortTokensByRelevance = (tokens: SearchAsset[], query: string): SearchItemWithRelevance[] => {
+  const normalizedQuery = query.toLowerCase().trim();
+  const tokenWithRelevance: SearchItemWithRelevance[] = tokens.map(token => {
+    const normalizedTokenName = token.name.toLowerCase();
+
+    const normalizedTokenSymbol = token.symbol.toLowerCase();
+    const tokenNameWords = normalizedTokenName.split(' ');
+    const relevance = getTokenRelevance({
+      token,
+      normalizedTokenName,
+      normalizedQuery,
+      normalizedTokenSymbol,
+      tokenNameWords,
+    });
+    return { ...token, relevance };
+  });
+
+  return tokenWithRelevance.sort((a, b) => b.relevance - a.relevance);
+};
+
+// higher number indicates higher relevance
+const getTokenRelevance = ({
```

Comment:
Wdyt about another +1 whenever a token has an `icon_url`? It would prevent ties like this:

![Screenshot 2025-03-31 at 7 18 34 PM](https://github.com/user-attachments/assets/d119ea90-628f-4d84-9663-310fd42412c7)

Reply:
<img width="437" alt="Screenshot 2025-04-01 at 2 00 30 PM" src="https://github.com/user-attachments/assets/fddf5cb5-98a5-4029-90cf-b0775d569f34" />